spring-cloud-dataflow-server-cloudfoundry-autoconfig/src/main/java/org/springframework/cloud/dataflow/server/cloudfoundry/config/DataSourceCloudConfig.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.server.cloudfoundry.config;
+
+import javax.sql.DataSource;
+
+import org.springframework.cloud.config.java.AbstractCloudConfig;
+import org.springframework.cloud.service.PooledServiceConnectorConfig;
+import org.springframework.cloud.service.relational.DataSourceConfig;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+/**
+ * Add declarative configuration of the max pool side and max wait time when using the connector library.
+ *
+ * @author Mark Pollack
+ */
+@Profile("cloud")
```

Comment:
I see some issues with the SCDF CF server deployment into PWS.
I wonder if this is related this specific change (and what auto-reconfiguration does):

```
 2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 2018-06-05 11:45:24.697 ERROR 14 --- [           main] o.s.boot.SpringApplication               : Application startup failed
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration': Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dataSource' defined in class path resource [org/springframework/cloud/dataflow/server/cloudfoundry/config/DataSourceCloudConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [javax.sql.DataSource]: Factory method 'dataSource' threw exception; nested exception is org.springframework.cloud.CloudException: No unique service matching interface javax.sql.DataSource found. Expected 1, found 0
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:749) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:189) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1197) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1099) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:513) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:312) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:308) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:372) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1177) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1071) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:513) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:312) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:308) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1080) ~[spring-context-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:857) ~[spring-context-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:543) ~[spring-context-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122) ~[spring-boot-1.5.12.RELEASE.jar!/:1.5.12.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693) [spring-boot-1.5.12.RELEASE.jar!/:1.5.12.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360) [spring-boot-1.5.12.RELEASE.jar!/:1.5.12.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:303) [spring-boot-1.5.12.RELEASE.jar!/:1.5.12.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1118) [spring-boot-1.5.12.RELEASE.jar!/:1.5.12.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1107) [spring-boot-1.5.12.RELEASE.jar!/:1.5.12.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.cloud.dataflow.server.cloudfoundry.CloudFoundryDataFlowServer.main(CloudFoundryDataFlowServer.java:34) [classes/:na]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_172]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_172]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_172]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_172]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) [app/:na]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) [app/:na]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) [app/:na]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51) [app/:na]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dataSource' defined in class path resource [org/springframework/cloud/dataflow/server/cloudfoundry/config/DataSourceCloudConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [javax.sql.DataSource]: Factory method 'dataSource' threw exception; nested exception is org.springframework.cloud.CloudException: No unique service matching interface javax.sql.DataSource found. Expected 1, found 0
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:599) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1177) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1071) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:513) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:312) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:308) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:208) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1138) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1066) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:835) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:741) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	... 36 common frames omitted
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [javax.sql.DataSource]: Factory method 'dataSource' threw exception; nested exception is org.springframework.cloud.CloudException: No unique service matching interface javax.sql.DataSource found. Expected 1, found 0
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:189) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	... 49 common frames omitted
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT Caused by: org.springframework.cloud.CloudException: No unique service matching interface javax.sql.DataSource found. Expected 1, found 0
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.cloud.Cloud.getSingletonServiceConnector(Cloud.java:197) ~[spring-cloud-connectors-core-2.0.2.RELEASE.jar!/:na]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.cloud.config.java.CloudServiceConnectionFactory.dataSource(CloudServiceConnectionFactory.java:56) ~[spring-cloud-spring-service-connector-2.0.2.RELEASE.jar!/:na]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.cloud.dataflow.server.cloudfoundry.config.DataSourceCloudConfig.dataSource(DataSourceCloudConfig.java:42) ~[spring-cloud-dataflow-server-cloudfoundry-autoconfig-1.5.1.BUILD-SNAPSHOT.jar!/:1.5.1.BUILD-SNAPSHOT]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.cloud.dataflow.server.cloudfoundry.config.DataSourceCloudConfig$$EnhancerBySpringCGLIB$$2a54273c.CGLIB$dataSource$0(<generated>) ~[spring-cloud-dataflow-server-cloudfoundry-autoconfig-1.5.1.BUILD-SNAPSHOT.jar!/:1.5.1.BUILD-SNAPSHOT]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.cloud.dataflow.server.cloudfoundry.config.DataSourceCloudConfig$$EnhancerBySpringCGLIB$$2a54273c$$FastClassBySpringCGLIB$$7388b951.invoke(<generated>) ~[spring-cloud-dataflow-server-cloudfoundry-autoconfig-1.5.1.BUILD-SNAPSHOT.jar!/:1.5.1.BUILD-SNAPSHOT]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228) ~[spring-core-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:358) ~[spring-context-4.3.16.RELEASE.jar!/:4.3.16.RELEASE]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.cloud.dataflow.server.cloudfoundry.config.DataSourceCloudConfig$$EnhancerBySpringCGLIB$$2a54273c.dataSource(<generated>) ~[spring-cloud-dataflow-server-cloudfoundry-autoconfig-1.5.1.BUILD-SNAPSHOT.jar!/:1.5.1.BUILD-SNAPSHOT]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_172]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_172]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_172]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_172]
   2018-06-05T17:15:24.69+0530 [APP/PROC/WEB/0] OUT 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162) ~[spring-beans-4.3.16.RELEASE.jar!/:4.3.
```

Reply:
I didn't see that.  Something non-deterministic is going on here.  Pulling this change out until we can investigate more.

Reply:
Turns out a service wasn't bound to the app, so nothing to worry about here